pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/knative/build/pkg/apis/build/v1alpha1"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// MountName is the name of the pvc being mounted (which
+	// will contain the entrypoint binary and eventually the logs)
+	MountName        = "tools"
+	MountPoint       = "/tools"
+	BinaryLocation   = MountPoint + "/entrypoint"
+	JSONConfigEnvVar = "ENTRYPOINT_OPTIONS"
+	Image            = "gcr.io/k8s-prow/entrypoint@sha256:7c7cd8906ce4982ffee326218e9fc75da2d4896d53cabc9833b9cc8d2d6b2b8f"
+	InitContainerName = "place-tools"
+	ProcessLogFile = "/tools/process-log.txt"
+	MarkerFile = "/tools/marker-file.txt"
+)
+
+var toolsMount = corev1.VolumeMount{
+	Name:      MountName,
+	MountPath: MountPoint,
+}
+
+type Cache struct {
+	mtx sync.RWMutex
+	cache map[string][]string
+}
+
+func NewCache() *Cache {
+	return &Cache{
+		cache: make(map[string][]string),
+	}
+}
+
+func (c *Cache) get(sha string) ([]string, bool) {
+	c.mtx.RLock()
+	ep, ok := c.cache[sha]
+	c.mtx.RUnlock()
+	return ep, ok
```

Comment:
could you explain a bit about why we need to protect this cache with a mutex? as far as I can tell, there isn't currently any need, but maybe are you planning to add some logic to retrieve the images in parallel? (or maybe it's already there and I'm not seeing it?)

my personal preference would be to avoid adding additional complication if we don't need it yet, but maybe it makes sense to be better safe than sorry in this case.

p.s. if this _did_ become a problem, i.e. if we didn't have the mutex but later it turned out we needed it, is this something [the race condition detector](https://blog.golang.org/race-detector) would catch?

Reply:
or is the idea that we want to persist this so that it can be used by multiple Reconcile loops simultaneously?

(might be worth adding some comments to explain this if so, and maybe a follow up issue?)

Reply:
Previously the cache was stored inside the reconciler and shared across workers. I left it in place since it was only a few lines of code, and then it is very simple to re-enable the caching once image digest resolving is done at a higher level as discussed in slack

I can add comments or remove the code entirely

Reply:
my preference is to remove the lock if it isn't needed, but im okay with leaving it if you'd like with a comment explaining why its there